examples/lock-app/qpg/src/BoltLockManager.cpp

```diff
@@ -173,14 +174,28 @@ bool BoltLockManager::SetCredential(uint16_t credentialIndex, FabricIndex creato
     return true;
 }
 
-bool BoltLockManager::ValidatePIN(const Optional<ByteSpan> & pinCode, OperationErrorEnum & err) const
+bool BoltLockManager::ValidatePIN(const Optional<ByteSpan> & pinCode, OperationErrorEnum & err, chip::EndpointId endpointId) const
 {
-    // Optionality of the PIN code is validated by the caller, so assume it is OK not to provide the PIN code.
+    // Assume pin is required until told otherwise
+    bool requirePin = true;
+    chip::app::Clusters::DoorLock::Attributes::RequirePINforRemoteOperation::Get(endpointId, &requirePin);
```

Comment:
The cluster logic itself enforces this, no?  Why does the application need to do this?

Reply:
You are correct, the behavior gets enabled when COTA feature and PIN feature is selected. I will update and let the doorlock server implementation handle it.